slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/DbSlashingProtection.java

```diff
@@ -36,12 +48,40 @@ public boolean maySignAttestation(
 
   @Override
   public boolean maySignBlock(
-      final String publicKey, final Bytes signingRoot, final UInt64 blockSlot) {
-    return true;
+      final Bytes publicKey, final Bytes signingRoot, final UInt64 blockSlot) {
+    return jdbi.inTransaction(
```

Comment:
confirming this is atomic - i.e. another signing can't "interlace" this transaction?
Does a 'big' transaction like this affect other threads accessing the attestation table?
I.e. do we need to mimise the lock size?

Reply:
It is atomic across transactions. Using serializable transaction isolation level https://www.postgresql.org/docs/12/transaction-iso.html#XACT-SERIALIZABLE in postgresql. This won't affect other tables other than the tables in the transaction. We should minimise the lock size and so have take the retrieve validators out of this transaction.